tests/test__common.py

```diff
@@ -10,3 +13,38 @@ def test__valid_toolbox_root_dir():
     assert path.is_absolute()
     # test that we can find this file using the toolbox root directory
     assert path / "tests" / "test__common.py" == Path(__file__)
+
+
+def test__import_namespaces():
+    def check_subpackage_imports(subpackage: ModuleType, directory_path: Path) -> None:
+        """Recursively asserts that all files/directories within a directory path are importable
+        from the subpackage
+
+        Args:
+            subpackage: The subpackage to check import from
+            directory_path: The directory to compare with the subpackage imports
+        """
+        print(f"Checking all files/directories within {directory_path} are importable")
+        # Collect names of packages we expect based on filenames in the directory
+        modules_by_files = {
+            f.replace(".py", "") for f in os.listdir(directory_path) if not f.startswith("_")
+        }
+        # Collect names of packages that are importable from the module
+        modules_by_namespace = set(subpackage.__all__)
+
+        # We use subset here to allow for imports that are not in the direct file system folder
+        # such as the project._common files which are imported through specific project types
+        assert modules_by_files.issubset(modules_by_namespace)
```

Comment:
This line only checks that the modules are in the manually created `__all__` attribute.  Checking that they have all been imported requires inspecting the result of the `__dir__()` method.
```suggestion
        assert modules_by_files.issubset(modules_by_namespace)
        assert modules_by_files.issubset(subpackage.__dir__())
```